### ISSUE TITLE
Support `hyphens: manual` for HTML output?

### ISSUE THREAD

**Issue by matklad** [asciidoctor/asciidoctor#3937]
Hi!

Someone [recently complained](https://news.ycombinator.com/item?id=26049547) that my asciidoctor-powered blog looks ugly due to the lack of proper hyphenation, and they have a point. In theory, hyphenation should be handled by the browser, but, in practice, browsers suck. `hyphens: auto` doesn't work for me in a chrome-derived browser on Linux, and produces ugly word breaks in firefox. This article from two years ago still seems appropriate: https://justmarkup.com/articles/2019-01-28-a-look-at-css-hyphenation-in-2019/

I wonder if asciidoctor should work-around this and use `hyphens: manual`, while pepper the output with `&shy;`s?

**Comment by matklad** [comment]
cc @Mogztter, I think this is especially pressing for asciidoctor-web-pdf, as pdf's are generally expected to have good typography.

**Comment by mojavelinux** [comment]
I don't think an AsciiDoc processor should be peppering output with `&shy;`. If this is something that you want to do in your own document, I recommend using a postprocessor. For my own documents, my personal opinion is that the browser should handle it.

**Comment by mojavelinux** [comment]
> produces ugly word breaks in firefox

Can you point to an example? We are using hyphenation on the docs.asciidoctor.org site and as far as I can see, it is working correctly.

**Comment by matklad** [comment]
> Can you point to an example?

Yeah, I am being unclear here, sorry. The problem for me is that the browsers break after only two letters, which personally seems excessive to me, and which I can't control via CSS

![image](https://user-images.githubusercontent.com/1711539/107145513-1f1b0a80-6953-11eb-81eb-68c1a0d8adc0.png)

>  my personal opinion is that the browser should handle it.

I totally agree with you, but the last time I checker the majority of browsers (chrome) didn't actually handle this. However, re-verifying this right now, it seem chrome did add support for `hyphen: auto` recently. I do get hyphenation in latest chrome, but I don't get it in Vivaldi.

**Comment by mojavelinux** [comment]
Thanks for those examples. It looks like more controls are coming in CSS4. See https://drafts.csswg.org/css-text-4/#hyphenate-size-limits. I think it's best to look towards the future. Hyphenation is a very complex subject and, again, not something I think the core processor should be taking on. If you need an immediate solution, you can use a postprocessor, as a previously suggested, or consider using a client-side library. You could also consider what Asciidoctor PDF is doing, which is sort of an integrated postprocessor. https://github.com/asciidoctor/asciidoctor-pdf/blob/master/lib/asciidoctor/pdf/converter.rb#L2941
